docs/source/getting_started/benchmarks/imagenette_benchmark.py

```diff
@@ -577,15 +591,95 @@ def configure_optimizers(self):
         return [optim], [scheduler]
 
 
+class MAEModel(BenchmarkModule):
+    def __init__(self, dataloader_kNN, num_classes):
+        super().__init__(dataloader_kNN, num_classes)
+        
+        decoder_dim = 512
+        vit = torchvision.models.vit_b_32(pretrained=False)
+
+        self.mask_ratio = 0.75
+        self.patch_size = vit.patch_size
+        self.sequence_length = vit.seq_length
+        self.mask_token = nn.Parameter(torch.zeros(1, 1, decoder_dim))
+        self.backbone = encoders.MAEBackbone.from_vit(vit)
+        self.decoder = encoders.MAEDecoder(
+            seq_length=vit.seq_length,
+            num_layers=1,
+            num_heads=16,
+            embed_input_dim=vit.hidden_dim,
+            hidden_dim=decoder_dim,
+            mlp_dim=decoder_dim * 4,
+            out_dim=vit.patch_size ** 2 * 3,
+            dropout=0,
+            attention_dropout=0,
+        )
+        self.criterion = nn.MSELoss()
+
+    def random_mask(self, images):
+        batch_size = images.shape[0]
+        idx_keep, idx_mask = utils.random_token_mask(
+            size=(batch_size, self.sequence_length),
+            mask_ratio=self.mask_ratio,
+            device=images.device,
+        )
+        return idx_keep, idx_mask
+
+    def forward_encoder(self, images, idx_keep=None):
+        return self.backbone.encode(images, idx_keep)
+
+    def forward_decoder(self, x_encoded, idx_keep, idx_mask):
+        # build decoder input
+        batch_size = x_encoded.shape[0]
+        x_decode = self.decoder.embed(x_encoded)
+        x_masked = utils.repeat_token(self.mask_token, (batch_size, self.sequence_length))
+        x_masked = utils.set_at_index(x_masked, idx_keep, x_decode)
+
+        # decoder forward pass
+        x_decoded = self.decoder.decode(x_masked)
+
+        # predict pixel values for masked tokens
+        x_pred = utils.get_at_index(x_decoded, idx_mask)
+        x_pred = self.decoder.predict(x_pred)
+        return x_pred
+
+    def training_step(self, batch, batch_idx):
```

Comment:
The length of the variables is always the batch size. If you convert all `len(x)` to `x.shape[1]` then your assumptions are correct :) I didn't add comments because it is only the benchmark module. The shapes are now well documented in the model/function docstrings. We could also add some more comments once we add the model examples.